src/dfx/src/lib/sns/deploy.rs

```diff
@@ -1,30 +1,54 @@
 //! Code for creating an SNS.
-use anyhow::bail;
+use anyhow::{anyhow, bail, Context};
 use fn_error_context::context;
 use std::ffi::OsString;
 use std::path::Path;
 
 use crate::lib::call_bundled::call_bundled;
 use crate::lib::error::DfxResult;
+use crate::lib::models::canister_id_store::CanisterIdStore;
+use crate::lib::provider::create_agent_environment;
 use crate::Environment;
 
 /// Creates an SNS.  This requires funds but no proposal.
 #[context("Failed to deploy SNS with config: {}", path.display())]
-pub fn deploy_sns(env: &dyn Environment, path: &Path) -> DfxResult<String> {
+pub fn deploy_sns(
+    env: &dyn Environment,
+    path: &Path,
+    network: Option<String>,
+) -> DfxResult<String> {
+    // For networks other than ic and local we need a provider URL:
+    let agent_environment = create_agent_environment(env, network.clone())?;
+    let network_descriptor = agent_environment.get_network_descriptor();
+    // The underlying sns binary recognises "local", "ic" and provider URLs, so we need one of those:
+    let sns_network_param = match network.as_ref().map(|network| network.as_ref()) {
+        None => "local",
+        Some("ic") => "ic",
+        Some("local") => "local",
+        Some(url) if url::Url::parse(url).is_ok() => url,
+        _ => network_descriptor.first_provider().with_context(|| {
```

Comment:
What's an example value that would meet the conditions asked for in the above code?
 1) doesn't parse as a valid url
 2) is then intuitively ignored in order to instead use the network descriptor's first provider
I guess I would expect the following, and no `_ =>` case
```
Some(url) -> url::Url::parse(url)?,
```